File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
  ans := make([]int, 0)
  var traverse func(t *TreeNode)
  traverse = func(t *TreeNode) {
    if t == nil {
      return
    }
    traverse(t.Left)
    ans = append(ans, t.Val)
    traverse(t.Right)
  }
  traverse(root)
  return ans
}

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
  const ans = []
  const dfs = t => {
    if (t == null) { return }
    dfs(t.left)
    ans.push(t.val)
    dfs(t.right)
  } 
  dfs(root)
  return ans
};